File: template_byTable/byTable/service/${className}Service.java
<#include "/java_copyright.include">
<#assign className = table.className>
<#assign classNameLower = className?uncap_first>
package ${basepackage}.service;


import ${basepackage}.model.${namespace}.${className};

import java.util.List;

/**
 * 
 * @Description: 业务层接口

 * @author: ???
 */
public interface ${className}Service {

	/**
	 *
	 * @Title: selectAll
	 * @Description: 查找所有记录
	 */
	List<${className}> selectAll();

	/**
	 *
	 * @Title: save

	 * @Description: 保存
	 */
	void save(${className} ${classNameLower});

	/**
	 *
	 * @Title: update

	 * @Description: 更新
	 */
	int update(${className} ${classNameLower});

<#list table.columns as column>
	<#if column.pk>
	/**
	 *
	 * @Title: selectByPK

	 * @Description: 查询
	 */
	${className} selectByPK(${column.javaType} id);

	/**
	 *
	 * @Title: delete
	 *
	 * @Description: 删除
	 */
	int delete(${column.javaType} id);
	</#if>
</#list>

}

File: template_byTable/${className}-ewandian-mybatis-postgresql/angular/${className}/services/${className}Service.js
<#include "/macro.include"/>
<#assign className = table.className>
<#assign classNameLower = className?uncap_first>
'use strict';
/**
 * @author：Tian
 * @date: ${now?string('yyyy/MM/dd')}
 */
define([], function () {
    var customersFactory = function ($http,$q) {
        var serviceBase = '/${namespace}Service/${classNameLower}/',
            factory = {};

        /** 获取${className}列表 */
        factory.getPageList = function (vo) {
            var d = $q.defer();
            $http.post(serviceBase+'selectPageList',vo||{})
                .success(function (response) { d.resolve(response); })
                .error  (function (response) { d.reject(response);});
            return d.promise;
        }

        /** 获取${className}列表 */
        factory.getList = function (vo) {
            var d = $q.defer();
            $http.post(serviceBase+'selectList',vo||{})
                .success(function (response) { d.resolve(response); })
                .error  (function (response) { d.reject(response);});
            return d.promise;
        }

        /**
         * 保存${className}（包括新增，更新）
         * @param ${classNameLower}
         */
        factory.save = function (vo) {
            var d = $q.defer();
            $http.post(serviceBase+(vo['${table.columns[0].sqlName}'] ? 'updateOne' : 'insertOne'), vo)
                .success(function (response) { d.resolve(response); })
                .error  (function (response) { d.reject(response);});
            return d.promise;
        };

        /**
         * 删除${className}
         * @param id 被删除的${classNameLower}的ID
         */
        factory.remove = function (id) {
            var d = $q.defer();
            $http.post(serviceBase+'deleteById/' + id)
                .success(function (response) { d.resolve(response); })
                .error  (function (response) { d.reject(response);});
            return d.promise;
        };

        return factory;
    };

    angular.module('${basepackage}.${namespace}.${classNameLower}Service',[]).factory('${classNameLower}Service', customersFactory);
});


